src/mod/endpoints/mod_sofia/sofia.c

```diff
@@ -1513,9 +1513,16 @@ static void our_sofia_event_callback(nua_event_t event,
 		}
 	}
 
-
 	if (sofia_private && sofia_private != &mod_sofia_globals.destroy_private && sofia_private != &mod_sofia_globals.keep_private) {
 		if (!zstr(sofia_private->gateway_name)) {
+			if (event == nua_r_unregister && status != 401 && status == 407) {
+				switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(session), SWITCH_LOG_DEBUG, "Destroy handle after unregister.\n");
```

Comment:
I don't see session being initialize, this may cause a crash. Just add the gateway name instead